lib/hutch/config.rb

```diff
@@ -5,42 +5,75 @@
 module Hutch
   class UnknownAttributeError < StandardError; end
 
+  # Configuration settings, available everywhere
+  #
+  # There are defaults, which can be overridden by ENV variables prefixed by
+  # <tt>HUTCH_</tt>, and each of these can be overridden using the {.set}
+  # method.
+  #
+  # @example Configuring on the command-line
+  #   HUTCH_PUBLISHER_CONFIRMS=false hutch
   module Config
     require 'yaml'
-
-    STRING_KEYS = %w(mq_host
-                     mq_exchange
-                     mq_vhost
-                     mq_username
-                     mq_password
-                     mq_api_host).freeze
-
-    NUMBER_KEYS = %w(mq_port
-                     mq_api_port
-                     heartbeat
-                     channel_prefetch
-                     connection_timeout
-                     read_timeout
-                     write_timeout
-                     graceful_exit_timeout
-                     consumer_pool_size).freeze
-
-    BOOL_KEYS = %w(mq_tls
-                   mq_verify_peer
-                   mq_api_ssl
-                   autoload_rails
-                   daemonise
-                   publisher_confirms
-                   force_publisher_confirms
-                   enable_http_api_use
-                   consumer_pool_abort_on_exception).freeze
-
-    ALL_KEYS = (BOOL_KEYS + NUMBER_KEYS + STRING_KEYS).map(&:to_sym).freeze
+    STRING_KEYS = Set.new
+    NUMBER_KEYS = Set.new
+    BOOL_KEYS = Set.new
+
+    # Define a String user setting
+    # @!visibility private
+    def self.string_setting(name, _default_value)
+      STRING_KEYS << name
+    end
+
+    # Define a Number user setting
+    # @!visibility private
+    def self.number_setting(name, _default_value)
+      NUMBER_KEYS << name
+    end
+
+    # Define a Boolean user setting
+    # @!visibility private
+    def self.boolean_setting(name, _default_value)
+      BOOL_KEYS << name
+    end
+
+    string_setting :mq_host, '127.0.0.1'
+    string_setting :mq_exchange, 'hutch'
+    string_setting :mq_vhost, '/'
+    string_setting :mq_username, 'guest'
+    string_setting :mq_password, 'guest'
+    string_setting :mq_api_host, '127.0.0.1'
+
+    number_setting :mq_port, 5672
+    number_setting :mq_api_port, 15672
+    number_setting :heartbeat, 30
+    number_setting :channel_prefetch, 0
+    number_setting :connection_timeout, 11
+    number_setting :read_timeout, 11
+    number_setting :write_timeout, 11
+    number_setting :graceful_exit_timeout, 11
+    number_setting :consumer_pool_size, 1
+
+    boolean_setting :mq_tls, false
+    boolean_setting :mq_verify_peer, true
+    boolean_setting :mq_api_ssl, false
+    boolean_setting :autoload_rails, true
+    boolean_setting :daemonise, false
+    boolean_setting :publisher_confirms, false
+    boolean_setting :force_publisher_confirms, false
+    boolean_setting :enable_http_api_use, true
+    boolean_setting :consumer_pool_abort_on_exception, false
```

Comment:
Perhaps these DSL-like methods could help set up the default_settings in some way.